src/models/lstm/LSTM.cpp

```diff
@@ -0,0 +1,240 @@
+    auto row = data.begin();
+    std::advance(row, 1);
+    //Loop from first row to end of data
+    for(; row != data.end(); ++row)
+    {
+	for(int i = 0; i < (*row).size(); i++)
+	{   //row has var, mean, std_dev
+ 	    //read into map keyed on var name, param name
+    	    params[ (*row)[0] ]["mean"] = strtof( (*row)[1].c_str(), NULL);
+  	    params[ (*row)[0] ]["std_dev"] = strtof( (*row)[2].c_str(), NULL);
+	}
+    }
+
+    return params;
+}
+
+namespace lstm {
+
+    /** @TODO: Make option to construct model without an initial state.
+     *  Consider initializing the empty state with: 
+     *  lstm_model(config, model_params, make_shared<lstm_state>(lstm_state())) {}     
+     */
+
+    /**
+     * Constructor for model object based on model parameters only.
+     *
+     * @param model_params Model parameters lstm_params struct.
+     */
+    lstm_model::lstm_model(lstm_config config, lstm_params model_params)
+            : config(config), model_params(model_params),
+            device( torch::Device(torch::kCPU) )
+    {
+        // ********** Set fluxes to null for now: it is bogus until first call of run function, which initializes it
+        fluxes = nullptr;
+        //manually set torch seed for reproducibility
+        torch::manual_seed(0);
+        useGPU = config.useGPU && torch::cuda::is_available();
+        device = torch::Device( useGPU ? torch::kCUDA : torch::kCPU );
+
+        lstm_model::initialize_state(config.initial_state_path.c_str());
+ 
+        //FIXME need to initialize the model states by running a "warmup"
+        //so when the model is constructed, run it on N timesteps prior to prediction period
+        //OR read those states in and create the initial state tensors
+
+        std::cout<<"model_params.pytorch_model_path: " << config.pytorch_model_path;
+
+        model = torch::jit::load(config.pytorch_model_path);
+        model.to( device );
+        // Set to `eval` model (just like Python)
+        model.eval();
+
+        //FIXME what is the SUPPOSED to do?
+        torch::NoGradGuard no_grad_;
+
+        this->scale = read_scale_params(config.normalization_path);
+        this->fluxes = std::make_shared<lstm::lstm_fluxes>(lstm::lstm_fluxes());
+
+    }
+
+    /**
+     * Initialize LSTM Model State.
+     * Reads the initial state from a specified CSV file. This function
+     * might need to change if there is an option to initialize a blank
+     * state.
+     * @param initial_state_path 
+     * @return
+     */
+    void lstm_model::initialize_state(std::string initial_state_path)
+    {
+
+        vector<double> h_vec;
+        vector<double> c_vec;
+        CSVReader reader(initial_state_path);
+        auto data = reader.getData();
+        std::vector<std::string> header = data[0];
+        //Advance the iterator to the first data row (skip the header)
+        auto row = data.begin();
+        std::advance(row, 1);
+        //Loop form first row to end of data
```

Comment:
Should probably  do a  quick sanity check that row has the required number of columns before the loop to avoid an out of bounds error on bad input:  `if( row.size() != 2 ){ throw error; }`